src/dialog/Dialog.ts

```diff
@@ -65,6 +67,24 @@ export default class Dialog extends DialogBase<DialogProperties> {
 		!this.properties.modal && this._onCloseClick();
 	}
 
+	private _onKeyUp(event: KeyboardEvent) {
+		const { keyCode } = event;
+
+		if (keyCode === Keys.Escape) {
+			this._onCloseClick();
+		}
+	}
+
+	constructor() {
+		super();
+
+		const keyUpFunc = this._onKeyUp.bind(this);
+		document.addEventListener('onkeyup', keyUpFunc);
```

Comment:
I really don't like having to use a document-level handler for this. Is it possible instead to use the root node and to trap focus somehow?

Reply:
This is a similar approach to that used in `SlidePane`. I don't believe it's worthwhile adding `focus trapping` complications at this point before we have a focus management meta / mixin available. I could attach / detach the listener when the dialog is open / closed instead perhaps?

Reply:
Fair enough.

Reply:
I think it's fine to have the handler on the root of the modal. It's not keyboard accessible now anyway without focus trapping, so I don't think it's worth the document-level handler for this.

Reply:
@smhigley  i couldn't get it to trigger the callback on the root of the model, but i will give it another go to make sure i did not mess it up.

Reply:
@tomdye it's a little unfortunate in that the focus doesn't move into the modal when it opens right now :/. I made my stuff comments rather than requested changes b/c I would totally understand if you decide to continue with the current approach so it just works.

Reply:
not a problem, i guess that this is something that we can address when focus management becomes a thing.